src/main/java/org/folio/helper/CheckinHelper.java

```diff
@@ -217,13 +218,23 @@ protected Future<Boolean> receiveInventoryItemAndUpdatePiece(JsonObject item, Pi
       })
       // Add processing error if item failed to be updated
       .onFailure(e -> {
-        logger.error("Item associated with piece '{}' cannot be updated", piece.getId());
-        addError(piece.getPoLineId(), piece.getId(), ITEM_UPDATE_FAILED.toError());
+        produceErrorMessage(piece, e.toString());
         promise.complete(false);
       });
     return promise.future();
   }
 
+  // This function produces error message based on the error it receives from the checkinItem()
+  protected void produceErrorMessage(Piece piece, String error) {
+    if (error.contains("Barcode must be unique")) {
```

Comment:
seems we don't have more clear way to handle this instead of relying on error message beased on this search results:
https://github.com/search?q=org%3Afolio-org+%22Barcode+must+be+unique%22&type=code&p=1
We can just move this string to constant
private static final String BARCODE_NOT_UNIQUE_MESSAGE = "Barcode must be unique";

Reply:
sure, will do.